pisa/analysis/analysis.py

```diff
@@ -2681,6 +2682,136 @@ def _minimizer_callback(self, xk, **unused_kwargs): # pylint: disable=unused-arg
         """
         self._nit += 1
 
+    def MCMC_sampling(self, data_dist, hypo_maker, metric, nwalkers, burnin, nsteps,
+                      return_burn_in=False, random_state=None):
+        """Performs MCMC sampling. Only supports serial (single CPU) execution at the 
+        moment. See issue #830.
+
+        Parameters
+        ----------
+
+        data_dist : Sequence of MapSets or MapSet
+            Data distribution to be fit. Can be an actual-, Asimov-, or pseudo-data
+            distribution (where the latter two are derived from simulation and so aren't
+            technically "data").
+
+        hypo_maker : Detectors or DistributionMaker
+            Creates the per-bin expectation values per map based on its param values.
+            Free params in the `hypo_maker` are modified by the minimizer to achieve a
+            "best" fit.
+
+        metric : string or iterable of strings
+            Metric by which to evaluate the fit. See documentation of Map.
+
+        nwalkers : int
+            Number of walkers
+
+        burnin : int
+            Number of steps in burn in phase
+
+        nSteps : int
+            Number of steps after burn in
+            
+        return_burn_in : bool
+            Also return the steps of the burn in phase. Default is False.
+
+        random_state : None or type accepted by utils.random_numbers.get_random_state
+            Random state of the walker starting points. Default is None.
+
+        Returns
+        -------
+
+        scaled_chain : numpy array
+            Array containing all points in the parameter space visited by each walker.
+            It is sorted by steps, so all the first steps of all walkers come first.
+            To for example get all values of the Nth parameter and the ith walker, use
+            scaled_chain[i::nwalkers, N].
+
+        scaled_chain_burnin : numpy array (optional)
+            Same as scaled_chain, but for the burn in phase.
+
+        """
+        import emcee
```

Comment:
May want to add a comment specifying which sampling algorithm the emcee package uses, for future reference.

Reply:
I added the option to define the sampling algorithm and pointed to the emcee docs where one can find the different options and the default